src/IO/Base.agda

```diff
@@ -128,3 +129,8 @@ untilJust : IO (Maybe A) → IO A
 untilJust m = bind (♯ m) λ where
   nothing  → ♯ untilJust m
   (just a) → ♯ return a
+
+untilRight : (A → IO (A ⊎ B)) → A → IO B
```

Comment:
Ah, I guess you can't get away with not having `A B` explicitly bound because then they're introduced
at distinct set levels and so the `IO` in the assumption and the one in the conclusion are not at compatible
levels.

You could either have
```suggestion
untilRight : {A B : Set a} → (A → IO (A ⊎ B)) → A → IO B
```
or
```suggestion
untilRight : {A : Set a} → (A → IO (A ⊎ B)) → A → IO (Lift a B)
```